src/Windows/R/Components/Test/InteractiveWorkflow/RInteractiveEvaluatorTest.cs

```diff
@@ -106,30 +103,35 @@ public class RInteractiveEvaluatorTest : IAsyncLifetime {
                 window.Operations.ClearView();
                 var result = await eval.ExecuteCodeAsync("w <- dQuote('text')" + Environment.NewLine);
                 result.Should().Be(ExecutionResult.Success);
-                window.FlushOutput();
-                var text = window.OutputBuffer.CurrentSnapshot.GetText();
+                await FlushOutputAsync(window);
 
                 window.Operations.ClearView();
                 result = await eval.ExecuteCodeAsync("w" + Environment.NewLine);
                 result.Should().Be(ExecutionResult.Success);
-                window.FlushOutput();
-                text = window.OutputBuffer.CurrentSnapshot.GetText();
+                await FlushOutputAsync(window);
+                var text = window.OutputBuffer.CurrentSnapshot.GetText();
                 text.TrimEnd().Should().Be("[1] \"“text”\"");
 
                 window.Operations.ClearView();
                 result = await eval.ExecuteCodeAsync("e <- dQuote('абвг')" + Environment.NewLine);
                 result.Should().Be(ExecutionResult.Success);
-                window.FlushOutput();
+                await FlushOutputAsync(window);
                 text = window.OutputBuffer.CurrentSnapshot.GetText();
                 text.Should().Be(string.Empty);
 
                 window.Operations.ClearView();
                 result = await eval.ExecuteCodeAsync("e" + Environment.NewLine);
                 result.Should().Be(ExecutionResult.Success);
-                window.FlushOutput();
+                await FlushOutputAsync(window);
                 text = window.OutputBuffer.CurrentSnapshot.GetText();
                 text.TrimEnd().TrimEnd((char)0).Should().Be("[1] \"“абвг”\"");
             }
         }
+
+        private async Task FlushOutputAsync(IInteractiveWindow window) {
+            // Give interactive window writer time to process message queue
+            await Task.Delay(400);
```

Comment:
This may fail with different test run configuration, which isn't determent. Maybe you can use `UIThreadHelper.Instance.DoEventsAsync` here?